src/python/grpcio_tests/tests/unit/_auth_context_test.py

```diff
@@ -0,0 +1,157 @@
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""Tests exposure of SSL auth context"""
+
+import pickle
+import unittest
+
+import grpc
+from grpc import _channel
+from grpc.framework.foundation import logging_pool
+
+from tests.unit import test_common
+from tests.unit.framework.common import test_constants
+from tests.unit import resources
+
+_REQUEST = b'\x00\x00\x00'
+_RESPONSE = b'\x00\x00\x00'
+
+_UNARY_UNARY = '/test/UnaryUnary'
+
+_SERVER_HOST_OVERRIDE = 'foo.test.google.fr'
+_CLIENT_IDS = [
+    b'*.test.google.fr', b'waterzooi.test.google.be', b'*.test.youtube.com',
+    b'192.168.1.3'
+]
+_ID = 'id'
+_ID_KEY = 'id_key'
+_AUTH_CTX = 'auth_ctx'
+
+
+def handle_unary_unary(request, servicer_context):
+    return pickle.dumps({
+        _ID: servicer_context.peer_identity(),
+        _ID_KEY: servicer_context.peer_identity_key(),
+        _AUTH_CTX: servicer_context.auth_context()
+    })
+
+
+class AuthContextTest(unittest.TestCase):
+
+    def setUp(self):
+        self._server = None
+
+    def tearDown(self):
+        if self._server is not None:
+            self._server.stop(0)
+
+    def testInsecure(self):
+        self._server_pool = logging_pool.pool(test_constants.THREAD_CONCURRENCY)
+        handler = grpc.method_handlers_generic_handler('test', {
+            'UnaryUnary':
+            grpc.unary_unary_rpc_method_handler(handle_unary_unary)
+        })
+        self._server = grpc.server(
+            self._server_pool,
+            handlers=(handler,),
+            maximum_concurrent_rpcs=test_constants.THREAD_CONCURRENCY)
+        port = self._server.add_insecure_port('[::]:0')
+        self._server.start()
+        self._channel = grpc.insecure_channel('localhost:%d' % port)
+        response = self._channel.unary_unary(_UNARY_UNARY)(_REQUEST)
+
+        auth_data = pickle.loads(response)
+        self.assertIsNone(None, auth_data[_ID])
+        self.assertIsNone(None, auth_data[_ID_KEY])
+        self.assertEqual({}, auth_data[_AUTH_CTX])
+
+    def testSecureNoCert(self):
+        self._server_pool = logging_pool.pool(test_constants.THREAD_CONCURRENCY)
+        handler = grpc.method_handlers_generic_handler('test', {
+            'UnaryUnary':
+            grpc.unary_unary_rpc_method_handler(handle_unary_unary)
+        })
+        self._server = grpc.server(
+            self._server_pool,
+            handlers=(handler,),
+            maximum_concurrent_rpcs=test_constants.THREAD_CONCURRENCY)
+        port = self._server.add_secure_port(
+            '[::]:0',
+            grpc.ssl_server_credentials(
+                [(resources.private_key(), resources.certificate_chain())]))
```

Comment:
Never write a list literal where a tuple literal will do. (Can the value on this line be a module-scope constant?)